bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ErrorValue.java

```diff
@@ -64,7 +65,7 @@ public ErrorValue(BType type, String reason, Object details) {
 
     @Override
     public String stringValue() {
-        return reason + " " + details.toString();
+        return "error " + reason + " " + Optional.ofNullable(details).map(Object::toString).orElse("");
```

Comment:
space after reason is not applicable if the details record is empty.

Reply:
fixed